gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/mapper/GroupMapper.java

```diff
@@ -76,4 +77,45 @@ default String mapApplicationRole(Map<RoleScope, String> roles) {
         }
         return roles.get(RoleScope.APPLICATION);
     }
+
+    default List<Group> mapFromCoreList(List<io.gravitee.apim.core.group.model.Group> coreGroups) {
+        return Objects.isNull(coreGroups) ? null : coreGroups.stream()
+                .map(this::mapFromCore)
+                .collect(Collectors.toList());
+    }
+
+    default Group mapFromCore(io.gravitee.apim.core.group.model.Group coreGroup) {
+        if (Objects.isNull(coreGroup)) {
+            return null;
+        }
+
+        return new Group()
+                .id(coreGroup.getId())
+                .name(coreGroup.getName())
+                .createdAt(coreGroup.getCreatedAt() != null ? coreGroup.getCreatedAt().toOffsetDateTime() : null)
+                .updatedAt(coreGroup.getUpdatedAt() != null ? coreGroup.getUpdatedAt().toOffsetDateTime() : null)
+                .maxInvitation(coreGroup.getMaxInvitation())
+                .lockApiRole(coreGroup.isLockApiRole())
+                .lockApplicationRole(coreGroup.isLockApplicationRole())
+                .systemInvitation(coreGroup.isSystemInvitation())
+                .emailInvitation(coreGroup.isEmailInvitation())
+                .disableMembershipNotifications(coreGroup.isDisableMembershipNotifications())
+                .apiPrimaryOwner(coreGroup.getApiPrimaryOwner())
+                .eventRules(coreGroup.getEventRules() != null ? mapCoreGroupEventRules(coreGroup.getEventRules()) : null);
+    }
+
+    default List<GroupEvent> mapCoreGroupEventRules(List<GroupEventRule> eventRules) {
+        return Objects.isNull(eventRules) ? null : eventRules.stream()
+                .filter(Objects::nonNull)
+                .map(rule -> {
+                    try {
+                        return GroupEvent.fromValue(rule.event().name());
+                    } catch (IllegalArgumentException e) {
+                        logger.error("Unable to parse GroupEventRule: " + rule.event().name());
+                        return null;
+                    }
+                })
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
```

Comment:
```suggestion
Group mapFromCore(io.gravitee.apim.core.group.model.Group coreGroup);
List<Group> mapFromCore(List<io.gravitee.apim.core.group.model.Group> coreGroups);
GroupEvent map(GroupEventRule groupEventRule);
```

Mapstruct should be able to do all of the mapping heavy-lifting for you 👍

Reply:
done, thanks!